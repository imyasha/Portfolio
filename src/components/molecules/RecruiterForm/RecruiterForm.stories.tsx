import { RecruiterForm, RecruiterFormProps } from './RecruiterForm';

export default {
	title: 'molecules/RecruiterForm',
	component: RecruiterForm,
};

export const RecruiterFormStory = (args: RecruiterFormProps) => (
	<RecruiterForm {...args} />
);

RecruiterFormStory.storyName = 'RecruiterForm';
RecruiterFormStory.args = {
	handleSubmit: () => console.log('Submitted'),
	success: false,
	error: false,
};
