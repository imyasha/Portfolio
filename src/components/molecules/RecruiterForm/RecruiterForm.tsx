import * as Yup from 'yup';
import { Form, Formik } from 'formik';

import { Box } from 'Atoms/Box';
import { Button } from 'Atoms/Button';
import { Input } from 'Atoms/Input';
import { Select } from 'Atoms/Select';

export interface RecruiterFormProps {
	handleSubmit: (
		values: Record<string, string>,
		setSubmitting: (isSubmitting: boolean) => void,
		resetForm: () => void
	) => void;
	success: boolean;
	error: boolean;
}

export const RecruiterForm = ({
	handleSubmit,
	success,
	error,
}: RecruiterFormProps) => {
	const typeOptions = [
		'I am an in-house recruiter',
		'I work for a recruitment firm',
		'I am an independent recruiter',
		'None of the above',
	];
	const validationSchema = Yup.object({
		firstName: Yup.string().required('This field is required'),
		lastName: Yup.string().required('This field is required'),
		email: Yup.string()
			.email('Please enter a valid email address')
			.required('This field is required'),
		company: Yup.string(),
		type: Yup.mixed()
			.oneOf(typeOptions, 'Please select an option')
			.required('Please select an option'),
	});

	return (
		<Formik
			initialValues={{
				firstName: '',
				lastName: '',
				email: '',
				company: '',
				type: '',
			}}
			validationSchema={validationSchema}
			onSubmit={(values, { setSubmitting, resetForm }) => {
				handleSubmit(values, setSubmitting, resetForm);
			}}
		>
			{({ isSubmitting }) => (
				<Form role="form" className="mt-8">
					<Box>
						{success && (
							<div className="bg-green-100 px-4 py-2 text-green-600 ring-1 ring-green-600 font-bold rounded-md">
								Thanks for signing up to my mailing list! I&apos;ll reach out as
								soon as I&apos;m ready for my next position.
							</div>
						)}
						{error && (
							<div className="bg-red-100 px-4 py-2 text-red-600 ring-1 ring-red-600 font-bold rounded-md">
								Something went wrong, please try again!
							</div>
						)}
						<Input label="First Name" placeholder="John" id="firstName" />
						<Input label="Last Name" placeholder="Doe" id="lastName" />
						<Input label="Company" id="company" />
						<Input
							label="Email Address"
							placeholder="johndoe@gmail.com"
							id="email"
							type="email"
						/>
						<Select label="Who are you?" id="type" options={typeOptions} />
						<div className="mt-6">
							<Button
								type="submit"
								disabled={isSubmitting}
								onClick={(noop: unknown) => noop}
								className="group"
							>
								<div className="block headline group-hover:text-off-white">
									{isSubmitting ? 'Loading ...' : 'Sign up'}
								</div>
							</Button>
						</div>
					</Box>
				</Form>
			)}
		</Formik>
	);
};
