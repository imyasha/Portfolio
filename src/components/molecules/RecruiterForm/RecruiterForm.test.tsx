import { RecruiterForm } from '../RecruiterForm';
import { render } from '@testing-library/react';

describe('RecruiterForm', () => {
	it('renders correctly', () => {
		const { container } = render(
			<RecruiterForm handleSubmit={jest.fn} success={false} error={false} />
		);
		expect(container).toMatchSnapshot();
	});
});
