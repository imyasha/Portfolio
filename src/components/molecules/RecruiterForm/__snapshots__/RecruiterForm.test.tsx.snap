// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecruiterForm renders correctly 1`] = `
<div>
  <form
    action="#"
    class="mt-8"
    role="form"
  >
    <div
      class="md:w-10/12 px-8 py-6 rounded-3xl glass-light dark:glass-dark mb-6 drop-shadow-lg hover:drop-shadow-2xl transition-all"
    >
      <label
        class="flex flex-col font-bold mb-2 mt-6"
        for="firstName"
      >
        First Name
        <input
          class="px-4 py-2 rounded-md my-2 ring-1 ring-grey-400 dark:ring-grey-700"
          id="firstName"
          name="firstName"
          placeholder="John"
          type="text"
          value=""
        />
        <div
          class="font-sm font-normal text-red-600"
        />
      </label>
      <label
        class="flex flex-col font-bold mb-2 mt-6"
        for="firstName"
      >
        Last Name
        <input
          class="px-4 py-2 rounded-md my-2 ring-1 ring-grey-400 dark:ring-grey-700"
          id="lastName"
          name="lastName"
          placeholder="Doe"
          type="text"
          value=""
        />
        <div
          class="font-sm font-normal text-red-600"
        />
      </label>
      <label
        class="flex flex-col font-bold mb-2 mt-6"
        for="firstName"
      >
        Company
        <input
          class="px-4 py-2 rounded-md my-2 ring-1 ring-grey-400 dark:ring-grey-700"
          id="company"
          name="company"
          type="text"
          value=""
        />
        <div
          class="font-sm font-normal text-red-600"
        />
      </label>
      <label
        class="flex flex-col font-bold mb-2 mt-6"
        for="firstName"
      >
        Email Address
        <input
          class="px-4 py-2 rounded-md my-2 ring-1 ring-grey-400 dark:ring-grey-700"
          id="email"
          name="email"
          placeholder="johndoe@gmail.com"
          type="email"
          value=""
        />
        <div
          class="font-sm font-normal text-red-600"
        />
      </label>
      <label
        class="flex flex-col font-bold mb-2 mt-6"
        for="type"
      >
        Who are you?
        <select
          class="px-4 py-2 rounded-md my-2 ring-1 ring-grey-400 dark:ring-grey-700"
          id="type"
          name="type"
        >
          <option
            value="none"
          >
            Please select
          </option>
          <option
            value="I am an in-house recruiter"
          >
            I am an in-house recruiter
          </option>
          <option
            value="I work for a recruitment firm"
          >
            I work for a recruitment firm
          </option>
          <option
            value="I am an independent recruiter"
          >
            I am an independent recruiter
          </option>
          <option
            value="None of the above"
          >
            None of the above
          </option>
        </select>
        <div
          class="font-sm font-normal text-red-600"
        />
      </label>
      <div
        class="mt-6"
      >
        <button
          class="block gradient-border transition transition-300 rounded-md px-4 py-2 headline hover:gradient-bg hover:text-off-white group"
          type="submit"
        >
          <div
            class="block headline group-hover:text-off-white"
          >
            Sign up
          </div>
        </button>
      </div>
    </div>
  </form>
</div>
`;
